### ISSUE TITLE
Question: `update_grid_from_samples` function

### ISSUE THREAD

**Issue by hi-jin** [KindXiaoming/pykan#253]
Hello,

I'm trying to fully understand the update_grid_from_samples function in KANLayer.py, but I'm having trouble with a particular part of the code.

Specifically, I don't understand the purpose of sorting x along the batch axis. Here is the relevant code snippet:

```python
x = torch.einsum('ij,k->ikj', x, torch.ones(self.out_dim, ).to(self.device)).reshape(batch, self.size).permute(1, 0)
x_pos = torch.sort(x, dim=1)[0]
y_eval = coef2curve(x_pos, self.grid, self.coef, self.k, device=self.device)
```
Here, x has a shape of (self.in_dim * self.out_dim, batch).

My concern is that sorting x along the batch axis (dim=1) might result in losing the 'batch' information. I would appreciate any clarification on why sorting along this axis is necessary and how it impacts the functionality of the update_grid_from_samples function.

Thank you for your assistance.

**Comment by hi-jin** [comment]
Oh, I just have come to understand the purpose of the code and resolved my own question. Therefore, I am closing this issue.
